Team417_PowerPlay/src/main/java/org/firstinspires/ftc/team417_PowerPlay/AprilTagAutonomousInitDetectionExample.java

```diff
@@ -0,0 +1,164 @@
+        camera.openCameraDeviceAsync(new OpenCvCamera.AsyncCameraOpenListener()
+        {
+            @Override
+            public void onOpened() {
+
+                camera.startStreaming(CAMERA_WIDTH, CAMERA_HEIGHT,OpenCvCameraRotation.UPRIGHT);
+                camera.startStreaming(CAMERA_WIDTH, CAMERA_HEIGHT, OpenCvCameraRotation.UPRIGHT);
+
+            }
+
+            @Override
+            public void onError(int errorCode) { }
+        });
+
+        telemetry.setMsTransmissionInterval(50);
+
+        /*
+         * The INIT-loop:
+         * This REPLACES waitForStart!
+         */
+        while (!isStarted() && !isStopRequested()) {
+            ArrayList<AprilTagDetection> currentDetections = aprilTagDetectionPipeline.getLatestDetections();
+
+            if (currentDetections.size() != 0) {
+                boolean tagFound = false;
+
+                for (AprilTagDetection tag : currentDetections) {
+                    if (tag.id == LEFT || tag.id == MIDDLE || tag.id == RIGHT) {
+                        tagOfInterest = tag;
+                        tagFound = true;
+                        break;
+                    }
+                }
+
+                if (tagFound) {
+                    telemetry.addLine("Tag of interest is in sight!\n\nLocation data:");
+                    tagToTelemetry(tagOfInterest);
+                } else {
+                    telemetry.addLine("Don't see tag of interest :(");
+
+                    if (tagOfInterest == null) {
+                        telemetry.addLine("(The tag has never been seen)");
+                    } else {
+                        telemetry.addLine("\nBut we HAVE seen the tag before; last seen at:");
+                        tagToTelemetry(tagOfInterest);
+                    }
+                }
+            }
+            telemetry.update();
+            sleep(20);
+        }
+
+        /*
+         * The START command just came in: now work off the latest snapshot acquired
+         * during the init loop.
+         */
+
+        /* Update the telemetry */
+        if (tagOfInterest != null) {
+            telemetry.addLine("Tag snapshot:\n");
+            tagToTelemetry(tagOfInterest);
+            telemetry.update();
+        } else {
+            telemetry.addLine("No tag snapshot available, it was never sighted during the init loop :(");
+            telemetry.update();
+        }
+
+        /* Actually do something useful */
+        if (tagOfInterest == null || tagOfInterest.id == LEFT) {
+            //trajectory
+        } else if (tagOfInterest.id == MIDDLE) {
+            //trajectory
+        } else {
+            //trajectory
+        }
+
+        // You wouldn't have this in your autonomous, this is just to prevent the sample from ending
+        while (opModeIsActive()) {sleep(20);}
+        // When we implement the rest of autonomous we can delete line the while loop
+    }
+
+    void tagToTelemetry(AprilTagDetection detection)
+    {
```

Comment:
https://google.github.io/styleguide/javaguide.html#s4.1-braces
```suggestion
    void tagToTelemetry(AprilTagDetection detection) {
```